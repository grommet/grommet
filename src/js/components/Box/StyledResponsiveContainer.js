import styled from 'styled-components';
import { styledComponentsConfig } from '../../utils/styles';

export const StyledResponsiveContainer = styled.div.withConfig(
  styledComponentsConfig,
)`
  container-type: inline-size;
`;
