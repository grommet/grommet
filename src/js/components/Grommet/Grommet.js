import React, { forwardRef, useEffect, useMemo, useRef, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  deviceResponsive,
  getBreakpoint,
  normalizeColor,
  useForwardedRef,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';
import { RootsContext } from '../../contexts/RootsContext';
import { OptionsContext } from '../../contexts/OptionsContext';
import { format, MessageContext } from '../../contexts/MessageContext';
import defaultMessages from '../../languages/default.json';
import { GrommetPropTypes } from './propTypes';
import { AnalyticsProvider } from '../../contexts/AnalyticsContext';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const defaultOptions = {};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    options = defaultOptions,
    messages: messagesProp,
    onAnalytics,
    ...rest
  } = props;
  const { background, dir, themeMode, userAgent } = props;
  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    // if user provides specific menu alignment, we don't want
    // the defaults to be included at all (can cause issues with controlMirror)
    // override merged value with themeProp value
    if (
      themeProp &&
      themeProp.menu &&
      themeProp.menu.drop &&
      themeProp.menu.drop.align
    ) {
      delete nextTheme.menu.drop.align;
      nextTheme.menu.drop.align = themeProp.menu.drop.align;
    }
    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';

    if (
      themeMode === 'auto' &&
      typeof window !== 'undefined' &&
      window.matchMedia &&
      window.matchMedia('(prefers-color-scheme: dark)').matches
    ) {
      nextTheme.dark = true;
    }

    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  const messages = useMemo(() => {
    // combine the passed in messages, if any, with the default
    // messages and format function.
    const nextMessages = deepMerge(
      defaultMessages,
      messagesProp?.messages || {},
    );
    return {
      messages: nextMessages,
      format: (opts) => {
        const message = messagesProp?.format && messagesProp.format(opts);
        return typeof message !== 'undefined'
          ? message
          : format(opts, nextMessages);
      },
    };
  }, [messagesProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(document.body.clientWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  const grommetRef = useForwardedRef(ref);

  // track open FocusedContainers in a global array to manage
  // focus event listeners for trapFocus
  const roots = useRef([]);
  useEffect(() => {
    if (grommetRef.current) roots.current.push(grommetRef.current);
  }, [grommetRef]);
  const rootsContextValue = useMemo(() => ({ roots }), []);

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <RootsContext.Provider value={rootsContextValue}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <OptionsContext.Provider value={options}>
              <MessageContext.Provider value={messages}>
                <AnalyticsProvider onAnalytics={onAnalytics}>
                  <StyledGrommet full={full} {...rest} ref={grommetRef}>
                    {children}
                  </StyledGrommet>
                  {full && <FullGlobalStyle />}
                </AnalyticsProvider>
              </MessageContext.Provider>
            </OptionsContext.Provider>
          </ContainerTargetContext.Provider>
        </RootsContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';
Grommet.propTypes = GrommetPropTypes;

export { Grommet };
