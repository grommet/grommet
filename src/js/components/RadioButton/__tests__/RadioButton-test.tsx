import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import 'jest-styled-components';
import 'jest-axe/extend-expect';
import 'regenerator-runtime/runtime';
import { hpe } from 'grommet-theme-hpe';

import { Grommet } from '../../Grommet';
import { Box } from '../../Box';
import { RadioButton } from '..';

describe('RadioButton', () => {
  test('should have no accessibility violations', async () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton name="test" a11yTitle="test" />
      </Grommet>,
    );

    const results = await axe(container);
    expect(results).toHaveNoViolations();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('basic', () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton name="test" value="1" />
        <RadioButton id="test id" name="test" value="2" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('basic renders outside grommet wrapper', () => {
    const { container } = render(<RadioButton name="test" value="1" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('label', () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton label="test label" name="test" value="1" />
        <RadioButton label={<div>test label</div>} name="test" value="2" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('checked', () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton checked name="test" value="1" onChange={jest.fn()} />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('disabled', () => {
    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton disabled name="test" value="1" />
        <RadioButton disabled checked name="test" value="2" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('children', () => {
    const child = ({ checked }: { checked: boolean }) => (
      <Box pad="small" background={checked ? 'accent-1' : 'control'} />
    );

    const { container } = render(
      <Grommet theme={hpe}>
        <RadioButton name="test" value="1">
          {child}
        </RadioButton>
        <RadioButton checked name="test" value="2" onChange={jest.fn()}>
          {child}
        </RadioButton>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('label themed', () => {
    const customTheme = {
      radioButton: {
        font: {
          weight: 500,
        },
      },
    };
    const { container } = render(
      <Grommet theme={customTheme}>
        <RadioButton label="test" name="test" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('background-color check themed', () => {
    const customTheme = {
      radioButton: {
        check: {
          background: {
            color: 'red',
          },
        },
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <RadioButton name="test" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('background-color themed', () => {
    const customTheme = {
      radioButton: {
        background: {
          color: 'blue',
        },
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <RadioButton name="test" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('background-color themed symbolic', () => {
    const customTheme = {
      radioButton: {
        check: {
          background: {
            color: 'brand',
          },
        },
      },
    };

    const { container } = render(
      <Grommet theme={customTheme}>
        <RadioButton name="test" />
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders custom circle icon', () => {
    const CustomCircleIcon = ({ ...rest }) => (
      <svg data-testid="custom-radioButton-icon" {...rest} />
    );
    const customTheme = {
      radioButton: {
        icons: {
          circle: CustomCircleIcon,
        },
      },
    };

    const { container, getByTestId } = render(
      <Grommet theme={customTheme}>
        <RadioButton name="test" checked onChange={jest.fn()} />
      </Grommet>,
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('custom-radioButton-icon')).toBeDefined();
  });

  test('should apply a11yTitle or aria-label', () => {
    const { container, getByLabelText } = render(
      <Grommet theme={hpe}>
        <RadioButton name="test" a11yTitle="test" />
        <RadioButton name="test" aria-label="test-2" />
      </Grommet>,
    );

    expect(getByLabelText('test')).toBeTruthy();
    expect(getByLabelText('test-2')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
