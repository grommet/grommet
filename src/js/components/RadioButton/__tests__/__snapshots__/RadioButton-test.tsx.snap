// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton background-color check themed 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton background-color themed 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  background-color: blue;
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton background-color themed symbolic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton basic 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
        value="1"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
  <label
    class="c1"
    for="test id"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        id="test id"
        name="test"
        type="radio"
        value="2"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton basic renders outside grommet wrapper 1`] = `
.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c0 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c0:hover input:not([disabled])+div,
.c0:hover input:not([disabled])+span {
  border-color: #000000;
}

.c2 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c4 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c3 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<label
  class="c0"
>
  <div
    class="c1 "
  >
    <input
      class="c2"
      name="test"
      type="radio"
      value="1"
    />
    <div
      class="c3 c4"
    />
  </div>
</label>
`;

exports[`RadioButton checked 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px #7D4CDB;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c6 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  transform: scale(1);
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px #7D4CDB;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        checked=""
        class="c3"
        name="test"
        type="radio"
        value="1"
      />
      <div
        class="c4 c5"
      >
        <svg
          class="c6"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="6"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton children 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #7D4CDB;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  padding: 12px;
}

.c5 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #6FFFB0;
  color: #444444;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  padding: 12px;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

@media only screen and (max-width: 768px) {
  .c4 {
    padding: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    padding: 6px;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
        value="1"
      />
      <div
        class="c4"
      />
    </div>
  </label>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        checked=""
        class="c3"
        name="test"
        type="radio"
        value="2"
      />
      <div
        class="c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton disabled 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c6 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px #7D4CDB;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  opacity: 0.5;
  cursor: default;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
}

.c7 {
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  transform: scale(1);
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

@media only screen and (max-width: 768px) {
  .c6 {
    border: solid 2px #7D4CDB;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
    disabled=""
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        disabled=""
        name="test"
        type="radio"
        value="1"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
  <label
    class="c1"
    disabled=""
  >
    <div
      class="c2 "
    >
      <input
        checked=""
        class="c3"
        disabled=""
        name="test"
        type="radio"
        value="2"
      />
      <div
        class="c6 c5"
      >
        <svg
          class="c7"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="6"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton label 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c2 {
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
        value="1"
      />
      <div
        class="c4 c5"
      />
    </div>
    <span
      class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
    >
      test label
    </span>
  </label>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
        value="2"
      />
      <div
        class="c4 c5"
      />
    </div>
    <div>
      test label
    </div>
  </label>
</div>
`;

exports[`RadioButton label themed 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c6 {
  font-weight: 500;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c2 {
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
    <span
      class="c6"
    >
      test
    </span>
  </label>
</div>
`;

exports[`RadioButton renders custom circle icon 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px #7D4CDB;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px #7D4CDB;
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        checked=""
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      >
        <svg
          as=".StyledRadioButton__StyledRadioButtonIcon-sc-g1f6ld-3"
          data-testid="custom-radioButton-icon"
          theme="[object Object]"
        />
      </div>
    </div>
  </label>
</div>
`;

exports[`RadioButton should apply a11yTitle or aria-label 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        aria-label="test"
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        aria-label="test-2"
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;

exports[`RadioButton should have no accessibility violations 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c1 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c1:hover input:not([disabled])+div,
.c1:hover input:not([disabled])+span {
  border-color: #000000;
}

.c3 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c5 {
  transform: scale(1);
}

@media only screen and (max-width: 768px) {
  .c4 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <div
      class="c2 "
    >
      <input
        aria-label="test"
        class="c3"
        name="test"
        type="radio"
      />
      <div
        class="c4 c5"
      />
    </div>
  </label>
</div>
`;
